### DISCUSSION TITLE
OOM on running pulsar-perf

### DISCUSSION THREAD

**Discussion by Amiaozhou** [apache/pulsar#18930] [Q&A]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

OS：Linux version 3.10.0-1160.el7.x86_64 
Pulsar： apache-pulsar-2.10.1

### Minimal reproduce step

1）benchmark test：
bin/pulsar-perf produce -r 8000 -n 1 -threads 3 -c 10 -s 1024 persistent://my_tenant/my_tomcat_mulstag/threads-topic-08

### What did you expect to see?

always right send

### What did you see instead?

org.apache.pulsar.common.allocator.PulsarByteBufAllocator - Exiting JVM process for OOM error: failed to allocate 4194304 byte(s) of direct memory (used: 28630319104, max: 28631367680)
io.netty.util.internal.OutOfDirectMemoryError: failed to allocate 4194304 byte(s) of direct memory (used: 28630319104, max: 28631367680)

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
Cannot reproduce locally. I suspect it's about configuration issues or your local env restrictions. Moved to the Q&A form.

**Comment by Maslino** [comment]
What's your jdk version? I have the same problem with jdk1.8.0_181 and Pulsar 2.10.3.

update:
It's ok if client and server are in the same data center. If across geo region, OOM. So it's related to RTT?
